.github/workflows/update-interfaces.yml

```diff
@@ -0,0 +1,41 @@
+name: Update protobuf stubs
+
+on:
+  push
+
+jobs:
+  update_proto:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Copy files
+      run: |
+        SRC=./api-interfaces/gooseai
+        TGT=./src/stability_sdk/interfaces/gooseai
+        KIND=generation
+        cp $SRC/$KIND/${KIND}_pb2_grpc.py $TGT/$KIND/${KIND}_pb2_grpc.py
+        cp $SRC/$KIND/${KIND}_pb2.py $TGT/$KIND/${KIND}_pb2.py
+        #############################
+        # for backwards compatibility
+        #############################
+        cp ./api-interfaces/src/proto/generation.proto ./src/proto/generation.proto 
+        cp $SRC/$KIND/${KIND}_pb_service.d.ts ./src/js/${KIND}_pb_service.d.ts
+        cp $SRC/$KIND/${KIND}_pb_service.js ./src/js/${KIND}_pb_service.js
+        cp $SRC/$KIND/${KIND}_pb.d.ts ./src/js/${KIND}_pb.d.ts
+        cp $SRC/$KIND/${KIND}_pb.js ./src/js/${KIND}_pb.js
+    - name: Commit files
+      run: |
+        git config --local user.name "dmarx"
+        git add ./src/**/*.py
+        git add ./src/**/*.js
+        git add ./src/**/*.ts
+        git add ./src/**/*.proto
+        git commit -m "Updated protobuf files"
+    - name: Push change
+      uses: ad-m/github-push-action@master
```

Comment:
I also lean towards having actions create PRs instead of just pushing changes (unless they are pushing to a branch which is automated only).  So using an action like https://github.com/marketplace/actions/create-pull-request.

Reply:
Unless of course the plan is to not manually change the files under these paths, then I think the push could be ok as a result of a manual commit to update the submodule.

Reply:
yeah that was the idea. main is a protected branch so this can only be triggered on other branches. So it works, but also just typing that out makes me feel worse about it. not a huge fan of part of our CI depending on future us knowing that the main branch needs to stay protected. having it create a PR instead of pushing is probably a good idea. something something human intervention. go ahead and tag this PR as "changes requested"